packages/xrpl/src/client/connection.ts

```diff
@@ -448,9 +449,34 @@ export class Connection extends EventEmitter {
       )
       this.ws.removeAllListeners()
       this.ws = null
-      this.emit('disconnected', code)
-      // If this wasn't a manual disconnect, then lets reconnect ASAP.
-      if (code !== INTENTIONAL_DISCONNECT_CODE) {
+
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- undefined can be passed in.
+      if (code === undefined) {
```

Comment:
Will typescript allow you to add undefined as a possible parameter values to line 439?

Reply:
Turns out yep! - I previously tried to just change `code` to an optional value which failed because optional values have to be at the end. I didn't try setting both `code` and `reason` to optional though, which works! (oops :P)